src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -488,13 +488,21 @@ private void SetApplicationIcon()
         {
             if (this.Icon == null && !DynamoModel.IsTestMode)
             {
-                var icon = System.Drawing.Icon.ExtractAssociatedIcon(System.Reflection.Assembly.GetEntryAssembly().Location);
-                var bmp = icon.ToBitmap();
-                MemoryStream stream = new MemoryStream();
-                bmp.Save(stream, System.Drawing.Imaging.ImageFormat.Png);
-                stream.Seek(0, SeekOrigin.Begin);
-                PngBitmapDecoder pngDecoder = new PngBitmapDecoder(stream, BitmapCreateOptions.None, BitmapCacheOption.Default);
-                this.Icon = pngDecoder.Frames[0];
+                var applicationPath = Process.GetCurrentProcess().MainModule.FileName;
+                try
+                {
+                    var icon = System.Drawing.Icon.ExtractAssociatedIcon(applicationPath);
```

Comment:
@mmisol - does this result in the existing behavior for Revit icon loading?

Reply:
This gets the icon from the running application. Since the extension window does not have the standard window frame, one does not get an icon on it automatically. This is the way I found to actually get it, which as I tested works for Sandbox and Revit.

Reply:
@mmisol @mjkkirschner Looking at this code again. We may have also fixed a bug that Dynamo4C3D showing icon as ACAD. I bet with the old code, entry Assembly is ACAD since Civil3D is on top of that. I cant really test it right now before C3D move up to .Net48. But will check at Global launch.

Reply:
@QilongTang This code is very recent. It was added with the first PR of the extension window #11246. It was not working for Revit, which is the main reason I changed it.